### PR TITLE
opentrack: 2022.3.0 -> 2023.3.0

### LINKED ISSUE #251257: Update request: opentrack 2022.3.0 → 2023.2.0
- Package name: opentrack
- Latest released version: 2023.2.0
- Current version on the unstable channel: 2022.3.0
- Current version on the stable/release channel: 2022.3.0
- [X] Checked the [nixpkgs pull requests](https://github.com/NixOS/nixpkgs/pulls)

**Notify maintainers**  
~@makefu~ (sorry for the incorrect ping!)
@zaninime

release: https://github.com/opentrack/opentrack/releases/tag/opentrack-2023.2.0

-----

Note for maintainers: Please tag this issue in your PR.

### PR DESCRIPTION
**Pull request by zaninime**
## Description of changes

- Update to [2023.3.0](https://github.com/opentrack/opentrack/releases/tag/opentrack-2023.3.0)
- Add patch for a specific configuration, tracked by upstream PR: https://github.com/opentrack/opentrack/pull/1754, as it's not NixOS specific.

Fixes #251257.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by zaninime** [comment]
```sh
$ nix-build --no-out-link -E 'let pkgs = import <nixpkgs> { }; in pkgs.callPackage ./default.nix {}'
```

is not the right way of building packages in PRs. Assuming you're in the root of the cloned git repo of nixpkgs, the command to use is:

```sh
$ nix-build -A opentrack
```

or the equivalent with flakes:
```sh
$ nix build '.#opentrack'
```

**Comment by mksafavi** [comment]
> ```
> source> building '/nix/store/0ram1h913ij12wbb5nzp3ai0bf6vp5n5-source.drv'
> source> 
> source> trying https://developer.x-plane.com/wp-content/plugins/code-sample-generation/sample_templates/XPSDK401.zip
> source>   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
> source>                                  Dload  Upload   Total   Spent    Left  Speed
> source>   0   146    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
> source> curl: (56) The requested URL returned error: 404
> source> error: cannot download source from any mirror
> ```

The url is changed for that zip file.
[0001-fixed-xplane-xpsdk401-url.patch.txt](https://github.com/user-attachments/files/17447269/0001-fixed-xplane-xpsdk401-url.patch.txt)

Does the wine support work? 
On the older versions [that I saw on youtube], there used to be a freetrack mode. 
On the newer versions the same menu is under x-plane that has the wine related options. But I can't really get it to work. Or test that's it's working.

File: pkgs/applications/misc/opentrack/default.nix [hunk 1]
```diff
@@ -18,14 +18,14 @@
...
-  version = "2022.3.0";
+  version = "2023.3.0";
 
   aruco = callPackage ./aruco.nix {};
 
   # license.txt inside the zip file is MIT
   xplaneSdk = fetchzip {
-    url = "https://developer.x-plane.com/wp-content/plugins/code-sample-generation/sample_templates/XPSDK303.zip";
-    sha256 = "11wqjsr996c5qhiv2djsd55gc373a9qcq30dvc6rhzm0fys42zba";
+    url = "https://developer.x-plane.com/wp-content/plugins/code-sample-generation/sample_templates/XPSDK401.zip";
```
Comment (mksafavi) [inline pkgs/applications/misc/opentrack/default.nix:27]: the url should be
`+    url = "https://developer.x-plane.com/wp-content/plugins/code-sample-generation/sdk_zip_files/XPSDK401.zip";`

**Review by mksafavi — CHANGES_REQUESTED** [review]

**Comment by lmeunier** [comment]
Now that #349971 is merged, could we close this PR?

**Pull request closed without merging**